### PR TITLE
Backend Documentation (Admin module)

### PR DESCRIPTION
**Pull request by tanay13**
## Issue that this pull request solves

 Closes: #321

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by tanay13** [comment]
@jackfrost13 please review this

File: backend/docs/admin.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# <u>Backend Documentation</u>
+</br>
+
+### Admin module
+The Admin module is responsible for all the functions related to admin , for example : fetching particular admin data from the database , creating a new account , changing password etc. 
+Before executing every route function validation middleware gets executed which checks whether the input is in the given format or not . For this validation process a library called joi is used. 
+### API description
+- <code> <b> /admin </b> </code>
```
Comment (jackfrost13) [inline backend/docs/admin.md:8]: you can keep it like `/admin` [GET] such that you can avoid method : GET

File: backend/docs/admin.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - GET the list of all admins
+    - method : <b>GET</b>
+    - response body : 
+        | Parameter   |
```
Comment (jackfrost13) [inline backend/docs/admin.md:12]: Please add the full table, follow my comments from auth review

**Review by jackfrost13 — CHANGES_REQUESTED** [review]

**Comment by tanay13** [comment]
@jackfrost13  Please review i made the changes

File: backend/docs/admin.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        | username    | string, required ,unique |
+        | email       | string, required ,unique |
+        | contact     | string ,required |
+        | isSuperAdmin| boolean |
+  
+    - ![diagram](images/basic.jpg)
+    <br>
+
+- [ <b>POST</b> ] :<code> <b> /admin </b> </code>
+    - For creating admin account
```
Comment (Kajol-Kumari) [inline backend/docs/admin.md:24]: Please replace this line with `Create a new admin account`

File: backend/docs/admin.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        | lastName    | string , required|
+        | username    | string , required , unique|
+        | email       | string, required ,unique|
+        | contact     | string , required|
+
+    - ![diagram](images/postsuperadmin.jpg)
+<br>
+
+- [ <b>PUT</b> ] :<code> <b> /admin/password </b> </code>
+    - To change the existing user's password
```
Comment (Kajol-Kumari) [inline backend/docs/admin.md:39]: Please change this line to `Change the existing admin's Password`

**Review by Kajol-Kumari — APPROVED** [review]
@tanay13 Please do these minor changes and rest LGTM :+1:

**Comment by tanay13** [comment]
@Kajol-Kumari Please review i made the changes

File: backend/docs/admin.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# <u>Backend Documentation</u>
+</br>
+
+### Admin module
+The Admin module is responsible for all the functions related to admin , for example : fetching particular admin data from the database , creating a new account , changing password etc. 
+Before executing every route function validation middleware gets executed which checks whether the input is in the given format or not . For this validation process a library called joi is used. 
+### API description
+- [<b>GET</b> ] :<code> <b> /admin </b> </code>
```
Comment (jackfrost13) [inline backend/docs/admin.md:8]: Refer https://github.com/HITK-TECH-Community/Community-Website/pull/357#discussion_r571865409

**Review by jackfrost13 — CHANGES_REQUESTED** [review]

**Comment by tanay13** [comment]
@jackfrost13 Please review , i made the changes

**Review by jackfrost13 — APPROVED** [review]

**Comment by Kajol-Kumari** [comment]
Hey @tanay13 Just to confirm. You made this PR under SWOC?

**Comment by tanay13** [comment]
@Kajol-Kumari Yes both my PRs are under SWOC

**Pull request merged**